rmf_site_format/src/anchor.rs

```diff
@@ -42,7 +43,14 @@ impl From<[f32; 2]> for Anchor {
 }
 
 fn to_slice(p: &[f32]) -> &[f32; 2] {
-    p.try_into().expect("Wrong array size")
+    match p.try_into() {
```

Comment:
I think this is a case where we're taking a fallible code path when we don't really need to.

It seems the entire purpose of `to_slice` is to rework the translation inside of an anchor's pose into a `&[f32; 2]` so that the `[x, y]` position of an anchor can be _borrowed_ no matter which variant is being used for the anchor.

Realistically a `&[f32; 2]` has almost no benefit over a `[f32; 2]` (value instead of borrow), and we can always produce a `[f32; 2]` value without any possibility of an `Err`.

I think the right fix for this is to have `translation_for_category` return a `[f32; 2]` and eliminate the use of `to_slice` and/or replace it with something that produces a `[f32; 2]` for a given `&[f32; 3]`.